cmd/gcp-controller-manager/app/gke_approver_test.go

```diff
@@ -238,18 +264,20 @@ func TestHandle(t *testing.T) {
 					},
 				}, nil
 			})
-			approver := gkeApprover{
-				client: client,
-				validators: []csrValidator{
-					{
-						approveMsg: "tester",
-						permission: authorization.ResourceAttributes{Group: "foo", Resource: "bar", Subresource: "baz"},
-						recognize: func(opts GCPConfig, csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest) bool {
-							return c.recognized
-						},
-						validate: c.validate,
-					},
+			validator := csrValidator{
+				approveMsg: "tester",
+				permission: authorization.ResourceAttributes{Group: "foo", Resource: "bar", Subresource: "baz"},
+				recognize: func(opts GCPConfig, csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest) bool {
+					return c.recognized
 				},
+				validate: c.validate,
+			}
+			if c.preApproveHook != nil {
```

Comment:
Put this in the literal above. If c.preApproveHook is nil the field will just stay nil

Reply:
Done.